src/rust/ide/lib/enso-protocol/src/language_server/types.rs

```diff
@@ -340,6 +340,75 @@ pub struct TextEdit {
     pub text: String
 }
 
+impl TextEdit {
+    /// Compute an edit that represents the difference between the two given strings based on their
+    /// common pre- and postfix. This is an approximation of the diff between the two strings that
+    /// assumes that anythign between the common prefix and the common post-fix has changed.
+    ///
+    /// Example:
+    /// ```
+    /// # use enso_protocol::language_server::{TextEdit, Position, TextRange};
+    /// # use enso_data::text::TextLocation;
+    /// let source = "\n333<->12345\n";
+    /// let target = "\n333x12345\n";
+    ///
+    /// let diff = TextEdit::from_prefix_postfix_differences(source,target);
+    ///
+    /// let edit_start:TextLocation = Position{line:1,character:3}.into();
+    /// let edit_end:TextLocation   = Position{line:1,character:6}.into();
+    /// let edit_range:TextRange = (edit_start..edit_end).into();
+    /// assert_eq!(diff, TextEdit{range:edit_range, text:"x".to_string()});
+    /// ```
+    pub fn from_prefix_postfix_differences(source:&str, target:&str) -> TextEdit {
+        use enso_data::text::Index;
+        use enso_data::text::TextLocation;
+
+        let start_ix = utils::string::find_prefix_end_index(source, target);
+        let end_ix   = utils::string::find_postfix_start_index(source, target);
+
+        let source_start_index    = Index::new(start_ix);
+        let source_end_index      = Index::new(source.len()-end_ix);
```

Comment:
I've updated all the examples to check for byte/char consistency. 
It seems that the `TextEdit` and `Position` works on char indices. I'll check that this is accounted for before sending out the messages. This might actually be a bug in the current handling, too.